pkg/auth/http.go

```diff
@@ -25,9 +25,20 @@ func AuthMiddleware(backendRepo repository.BackendRepository) echo.MiddlewareFun
 				return next(c)
 			}
 
-			token, workspace, err := backendRepo.AuthorizeToken(c.Request().Context(), tokenKey)
+			var token *types.Token
+			var workspace *types.Workspace
+
+			token, workspace, err := workspaceRepo.AuthorizeToken(tokenKey)
 			if err != nil {
-				return echo.NewHTTPError(http.StatusUnauthorized)
+				token, workspace, err = backendRepo.AuthorizeToken(c.Request().Context(), tokenKey)
+				if err != nil {
+					return echo.NewHTTPError(http.StatusUnauthorized)
+				}
+
+				err = workspaceRepo.SetAuthorizationToken(token, workspace)
+				if err != nil {
+					return echo.NewHTTPError(http.StatusInternalServerError)
+				}
```

Comment:
I think there might be an opportunity to streamline this logic. 

Could we turn this into a function and pass the function into AuthMiddleware and NewAuthInterceptor? This way we have one place for this logic and neither one of these need repos.

Reply:
wouldn't the function need access to the repo still? So we will have to initialize the function in gateway and passing it down from Gateway -> abstraction -> authMiddleware

Reply:
Its the same as passing workspaceRepo down, but this would dedup the auth logic. 

Going to approve this since it'll works as-is, like before.

Reply:
Thanks. I feel like this makes more sense because it's consistent with the scheme of passing down repos as necessary. 

Also WorkspaceRepo might be used in abstractions in the future since it is a general user-specific cache